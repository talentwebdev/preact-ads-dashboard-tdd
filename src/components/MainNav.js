import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";
import Logo from "../assets/logo4.png";
import Logo2 from "../assets/proimg.jpg";
import Button from "@material-ui/core/Button";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import "../styles/MainNav.css";
import { FaBars, FaTimes } from "react-icons/fa";
import ArrowDropDownIcon from "@mui/icons-material/ArrowDropDown";

function MainNav() {
  const [showLinks, setShowLinks] = useState(false);
  const [anchorEl, setAnchorEl] = useState(null);

  const toggleNav = () => {
    setShowLinks(!showLinks);
  };

  const handleClick = (e) => {
    setAnchorEl(e.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };
  const [screenWidth, setScreenWidth] = useState(window.innerWidth);
  useEffect(() => {
    const changeWidth = () => {
      setScreenWidth(window.innerWidth);
    };
    window.addEventListener("resize", changeWidth);
    return () => {
      window.removeEventListener("resize", changeWidth);
    };
  });
  return (
    <div className="mainNav">
      <div className="logoSide">
        <div id="logoImg">
          <Link to="/">
            <img src={Logo} alt="BHHS" />
          </Link>{" "}
        </div>{" "}
      </div>{" "}
      <div className="leftSide">
        <div className="links" id={showLinks ? "hidden" : ""}>
          <Link onClick={toggleNav} className="items" to="/">
            Home{" "}
          </Link>{" "}
          <Link onClick={toggleNav} className="items" to="/products">
            Products{" "}
          </Link>{" "}
          <Link onClick={toggleNav} className="items" to="/newsroom">
            Newsroom{" "}
          </Link>{" "}
          <Link onClick={toggleNav} className="items" to="/contact">
            Contact{" "}
          </Link>{" "}
          <Link onClick={toggleNav} className="items" to="/faq">
            FAQ{" "}
          </Link>{" "}
          <Link onClick={toggleNav} className="items" to="/tutorial">
            Tutorial{" "}
          </Link>{" "}
          <Link onClick={toggleNav} className="items" to="/partners">
            Partners{" "}
          </Link>{" "}
        </div>{" "}
        {showLinks ? (
          <FaTimes onClick={toggleNav} className="btn" />
        ) : (
          <FaBars className="btn" onClick={toggleNav} />
        )}{" "}
      </div>{" "}
      <div className="rightSide">
        <div id="proImg">
          <img src={Logo2} alt="BHHS" onClick={handleClick} />{" "}
        </div>{" "}
        <div id="proMenu">
          <ArrowDropDownIcon className="arrowBtn" onClick={handleClick} />{" "}
          <p onClick={handleClick}> John Smith </p>{" "}
          <Menu
            id="simple-menu"
            anchorEl={anchorEl}
            keepMounted
            open={Boolean(anchorEl)}
            onClose={handleClose}
          >
            <MenuItem> Profile </MenuItem> <MenuItem> My account </MenuItem>{" "}
            <MenuItem> Logout </MenuItem>{" "}
          </Menu>{" "}
        </div>{" "}
      </div>{" "}
    </div>
  );
}

export default MainNav;
