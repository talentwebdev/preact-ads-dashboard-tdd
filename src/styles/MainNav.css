.mainNav{display:-webkit-box;display:-ms-flexbox;display:flex;height:50px;max-width:80%;margin:auto;background-color:#021E39}.mainNav .logoSide{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-flex:20%;-ms-flex:20%;flex:20%}.mainNav .logoSide #logoImg{width:100%;text-align:right;display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-pack:end;-ms-flex-pack:end;justify-content:flex-end;-webkit-box-align:center;-ms-flex-align:center;align-items:center}.mainNav .logoSide #logoImg img{margin-right:3vw;margin-left:20px;height:30px}.mainNav .leftSide{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-flex:45%;-ms-flex:45%;flex:45%;-webkit-box-pack:start;-ms-flex-pack:start;justify-content:flex-start;-webkit-box-align:center;-ms-flex-align:center;align-items:center;background-color:#021E39}.mainNav .leftSide .links{max-height:50px;width:100%}.mainNav .leftSide .links a{color:#fff;text-decoration:none;font-size:16px;margin-right:40px;font-weight:500;font-family:Arial, Helvetica, sans-serif}.mainNav .leftSide svg{color:#fff;font-size:25px;display:none}.mainNav .rightSide{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-flex:15%;-ms-flex:15%;flex:15%;background-color:#021E39}.mainNav .rightSide #proImg{width:100%;display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-ms-flex-pack:start;justify-content:flex-start;-webkit-box-align:left;-ms-flex-align:left;align-items:left;margin:0;padding:0}.mainNav .rightSide #proImg img{border-radius:3px;text-align:center;margin:5px;height:40px;cursor:pointer}.mainNav .rightSide #proMenu{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:horizontal;-webkit-box-direction:normal;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:start;-ms-flex-pack:start;justify-content:flex-start;-webkit-box-align:center;-ms-flex-align:center;align-items:center;width:100%;margin:0;padding:0}.mainNav .rightSide #proMenu p{color:#fff;cursor:pointer}.mainNav .rightSide #proMenu .arrowBtn{color:#fff;cursor:pointer}@media only screen and (max-width: 1150px){.mainNav .leftSide .links a{font-size:15px;margin-right:10px}}@media only screen and (max-width: 1050px){.mainNav .leftSide .links a{font-size:15px;margin-right:10px}}@media only screen and (max-width: 950px){.mainNav .leftSide .links a{font-size:14px;margin-right:10px}.rightSide #proImg img{height:35px}}@media only screen and (max-width: 850px){.mainNav .leftSide .links a{font-size:13px;margin-right:7px}.rightSide #proImg img{height:30px}}@media only screen and (max-width: 750px){.mainNav .leftSide .links a{font-size:12px;margin-right:5px}.rightSide #proImg img{height:30px}.rightSide #proImg{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center}}@media only screen and (max-width: 650px){.mainNav .leftSide .links a{font-size:11px;margin-right:2px}.rightSide #proImg{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding:0}.rightSide #proMenu p{padding:0px;font-size:10px}.rightSide #proMenu{display:none}}@media only screen and (max-width: 600px){.mainNav{max-width:100%}.mainNav .leftSide svg{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-pack:end;-ms-flex-pack:end;justify-content:flex-end;-webkit-box-align:center;-ms-flex-align:center;align-items:center;margin:auto}.mainNav .leftSide .links{display:none}.mainNav .leftSide .links .items:nth-child(1){margin-top:100px}.mainNav .leftSide #hidden{position:absolute;top:50px;background-color:#000;display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;max-height:600px;height:auto;left:0;width:100%;-webkit-transform:translateX(-100%, 60%);transform:translateX(-100%, 60%);-webkit-transition:all 0.5s ease;transition:all 0.5s ease}.mainNav .leftSide #hidden a{font-size:20px;padding:15px}.mainNav .leftSide #hidden a:hover{background-color:#f9f9f9;color:#000;font-weight:500;width:100%;text-align:center}.mainNav .rightSide #proImg{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center}.mainNav .rightSide #proMenu{display:none}}@media screen and (min-width: 650px){.rightSide #proImg{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center}.rightSide #proMenu p{padding:5px}}
