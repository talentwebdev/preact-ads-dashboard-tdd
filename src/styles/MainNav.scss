.mainNav {
    display: flex;
    height: 50px;
    max-width: 80%;
    margin: auto;
    background-color: #021E39;
    .logoSide {
        display: flex;
        flex: 20%;
        #logoImg {
            width: 100%;
            text-align: right;
            display: flex;
            justify-content: flex-end;
            align-items: center;
            img {
                margin-right: 3vw;
                margin-left: 20px;
                height: 30px;
            }
        }
    }
    .leftSide {
        display: flex;
        flex: 45%;
        justify-content: flex-start;
        align-items: center;
        background-color: #021E39;
        .links {
            max-height: 50px;
            width: 100%;
            a {
                color: #fff;
                text-decoration: none;
                font-size: 16px;
                margin-right: 40px;
                font-weight: 500;
                font-family: Arial, Helvetica, sans-serif;
            }
        }
        svg {
            color: #fff;
            font-size: 25px;
            display: none;
        }
    }
    .rightSide {
        display: flex;
        flex: 15%;
        background-color: #021E39;
        #proImg {
            width: 100%;
            display: flex;
            justify-content: flex-start;
            align-items: left;
            margin: 0;
            padding: 0;
            img {
                border-radius: 3px;
                text-align: center;
                margin: 5px;
                height: 40px;
                cursor: pointer;
            }
        }
        #proMenu {
            display: flex;
            flex-direction: row;
            justify-content: flex-start;
            align-items: center;
            width: 100%;
            margin: 0;
            padding: 0;
            p {
                color: #fff;
                cursor: pointer;
            }
            .arrowBtn {
                color: #fff;
                cursor: pointer;
            }
        }
    }
}

@media only screen and (max-width: 1150px) {
    .mainNav .leftSide .links a {
        font-size: 15px;
        margin-right: 10px;
    }
}

@media only screen and (max-width: 1050px) {
    .mainNav .leftSide .links a {
        font-size: 15px;
        margin-right: 10px;
    }
}

@media only screen and (max-width: 950px) {
    .mainNav .leftSide .links a {
        font-size: 14px;
        margin-right: 10px;
    }
    .rightSide #proImg img {
        height: 35px;
    }
}

@media only screen and (max-width: 850px) {
    .mainNav .leftSide .links a {
        font-size: 13px;
        margin-right: 7px;
    }
    .rightSide #proImg img {
        height: 30px;
    }
}

@media only screen and (max-width: 750px) {
    .mainNav .leftSide .links a {
        font-size: 12px;
        margin-right: 5px;
    }
    .rightSide #proImg img {
        height: 30px;
    }
    .rightSide #proImg {
        display: flex;
        justify-content: center;
        align-items: center;
    }
}

@media only screen and (max-width: 650px) {
    .mainNav .leftSide .links a {
        font-size: 11px;
        margin-right: 2px;
    }
    .rightSide #proImg {
        display: flex;
        justify-content: center;
        align-items: center;
        padding: 0;
    }
    .rightSide #proMenu p {
        padding: 0px;
        font-size: 10px;
    }
    .rightSide #proMenu {
        display: none;
    }
}

@media only screen and (max-width: 600px) {
    .mainNav {
        max-width: 100%;
        .leftSide {
            svg {
                display: flex;
                justify-content: flex-end;
                align-items: center;
                margin: auto;
            }
            .links {
                display: none;
                .items:nth-child(1) {
                    margin-top: 100px;
                }
            }
            #hidden {
                position: absolute;
                top: 50px;
                background-color: #000;
                display: flex;
                flex-direction: column;
                justify-content: center;
                align-items: center;
                max-height: 600px;
                height: auto;
                left: 0;
                width: 100%;
                transform: translateX(-100%, 60%);
                transition: all 0.5s ease;
                a {
                    font-size: 20px;
                    padding: 15px;
                }
                a:hover {
                    background-color: #f9f9f9;
                    color: #000;
                    font-weight: 500;
                    width: 100%;
                    text-align: center;
                }
            }
        }
        .rightSide {
            #proImg {
                display: flex;
                justify-content: center;
                align-items: center;
            }
            #proMenu {
                display: none;
            }
        }
    }
}

@media screen and (min-width: 650px) {
    .rightSide {
        #proImg {
            display: flex;
            justify-content: center;
            align-items: center;
        }
        #proMenu p {
            padding: 5px;
        }
    }
}