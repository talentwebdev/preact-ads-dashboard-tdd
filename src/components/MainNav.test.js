import {shallow} from 'enzyme';
import MainNav from './MainNav';

describe('MainNav', () => {
    let mainNavContainer;

    beforeEach(() => mainNavContainer = shallow(<MainNav />));

    it('should render the div element', () => {
        expect(mainNavContainer.find('div').length).toBeGreaterThanOrEqual(1);
    });

    it('should render the Link tag', () => {
        expect(mainNavContainer.find('Link').length).toBeGreaterThanOrEqual(1);
    });
});