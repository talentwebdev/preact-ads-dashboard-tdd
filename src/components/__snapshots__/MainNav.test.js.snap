// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNav renders MainNav correctly 1`] = `ShallowWrapper {}`;
